File: backend/internal/models/shoppingList.go
package models

type ShoppingList struct {
	ID uint `gorm:"primaryKey"`

	Name string `json:"name"`

	// Owner
	OwnerID uint `gorm:"not null"`
	Owner   User `gorm:"foreignKey:OwnerID;references:ID"`

	// Shared users (many to many)
	Users []User `gorm:"many2many:shopping_list_users;"`

	// Items and custom items
	Items       []Item
	CustomItems []CustomItem

	InviteToken string `json:"inviteToken"`
}

File: portal/src/models/ShoppingList.ts
import type { Item } from "./Item";
import type { User } from "./User";
import { buildItem } from "./Item"
import { buildUser } from "./User"

export type ShoppingList = {
  id: number;
  name: string;
  ownerID: number;
  owner: User;
  users: User[];
  items: Item[];
  customItems: Item[];
};


export function buildShoppingList(raw: any): ShoppingList {
  return {
    id: raw.ID,
    name: raw.name,
    ownerID: raw.OwnerID,
    owner: buildUser(raw.Owner),
    users: (raw.Users || []).map(buildUser),
    items: (raw.Items || []).map(buildItem),
    customItems: (raw.CustomItems || []).map(buildItem),
  }
}
